### PR TITLE
Bug 2039618: Add validations to SAP HANA

### PR DESCRIPTION
**Pull request by avivtur**
**Fixes**:
https://bugzilla.redhat.com/show_bug.cgi?id=2039618

**Analysis / Root cause**:
parameters were allowed to be empty

**Solution Description**:
 Add validations to ensure required parameters have value.
Restore initial parameter values in case of empty parameter values and add a placeholder `(generate if empty)` to empty non-required parameters which have an initial value.

Signed-off-by: Aviv Turgeman <aturgema@redhat.com>

### CODE REVIEW

**Comment by avivtur** [comment]
@yaacov please review

File: frontend/packages/kubevirt-plugin/src/components/vm-templates/instantiate-template/TemplateForm.tsx [hunk 1]
```diff
@@ -157,7 +187,10 @@ export const TemplateForm: React.FC<TemplateFormProps> = ({
             }: TemplateParameter) => {
               const paramValue = parameters[name] || '';
               const helpID = description ? `${name}-help` : '';
-              const placeholder = generate ? '(generated if empty)' : '';
+              const placeholder =
+                generate || (missingParams.find(([key]) => key === name) && !requiredParam)
```
Comment (yaacov) [inline frontend/packages/kubevirt-plugin/src/components/vm-templates/instantiate-template/TemplateForm.tsx:191]: help text should be "(set to <value> if empay)"

Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/vm-templates/instantiate-template/TemplateForm.tsx:191]: (set to if [value] empay)"

Comment (yaacov) [reply frontend/packages/kubevirt-plugin/src/components/vm-templates/instantiate-template/TemplateForm.tsx:191]: `(set to [value] if left empay)`

Comment (avivtur) [reply frontend/packages/kubevirt-plugin/src/components/vm-templates/instantiate-template/TemplateForm.tsx]: fixed

**Pull request merged**